File: release/src/router/libnl-bf/source/python/netlink/route/links/dummy.py
#
# Copyright (c) 2011 Thomas Graf <tgraf@suug.ch>
#

"""Dummy

"""

__version__ = "1.0"
__all__ = ['init']

import netlink.core as netlink
import netlink.route.capi as capi

class DummyLink(object):
	def __init__(self, link):
        	self._rtnl_link = link

	def brief(self):
        	return 'dummy'
	
def init(link):
	link.dummy = DummyLink(link._rtnl_link)
        return link.dummy

File: release/src-rt-5.02axhnd.675x/bcmdrivers/broadcom/net/wl/impl61/main/components/opensource/router_tools/libnl/lib/route/link/dummy.c
/*
 * lib/route/link/dummy.c	Dummy Interfaces
 *
 *	This library is free software; you can redistribute it and/or
 *	modify it under the terms of the GNU Lesser General Public
 *	License as published by the Free Software Foundation version 2.1
 *	of the License.
 *
 * Copyright (c) 2011 Thomas Graf <tgraf@suug.ch>
 */

/**
 * @ingroup link
 * @defgroup dummy Dummy
 *
 * @details
 * \b Link Type Name: "dummy"
 *
 * @{
 */

#include <netlink-private/netlink.h>
#include <netlink/netlink.h>
#include <netlink-private/route/link/api.h>

static struct rtnl_link_info_ops dummy_info_ops = {
	.io_name		= "dummy",
};

static void __init dummy_init(void)
{
	rtnl_link_register_info(&dummy_info_ops);
}

static void __exit dummy_exit(void)
{
	rtnl_link_unregister_info(&dummy_info_ops);
}

/** @} */
